File: UseExternalApps/app/src/main/java/com/example/useexternalapps/MainActivity.java
package com.example.useexternalapps;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    EditText phoneNumber;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        phoneNumber = findViewById(R.id.editPhoneNumber);
    }

    public void makeCall(View view) {
        Intent intentMakeCall = new Intent(Intent.ACTION_DIAL, Uri.parse("tel:" + phoneNumber.getText().toString()));
        startActivity(intentMakeCall);
    }

    public void goOnAboutPage(View view) {
        Intent intentAboutPage = new Intent(this, AboutActivity.class);
        startActivity(intentAboutPage);
    }
}

File: Lists_2/app/src/main/java/com/example/a12_lists/MainActivity.kt
package com.example.a12_lists

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.example.a12_lists.fragments.MainFragment
import com.example.a12_lists.fragments.OpenFragment

class MainActivity : AppCompatActivity(), OpenFragment {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            val mainFragment = MainFragment()
            supportFragmentManager.beginTransaction().replace(R.id.mainContainer, mainFragment)
                .setPrimaryNavigationFragment(mainFragment)
                .commit()
        }
    }

    override fun openFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction().replace(
            R.id.mainContainer,
            fragment
        )
            .addToBackStack("1")
            .commit()
    }


}
